src/wp-includes/functions.php

```diff
@@ -7830,6 +7830,29 @@ function wp_unique_id( $prefix = '' ) {
 	return $prefix . (string) ++$id_counter;
 }
 
+/**
+ * Generates an incremental ID that is independent per each different prefix.
+ *
+ * It is similar to `wp_unique_id`, but each prefix has its own internal ID
+ * counter to make each prefix independent from each other. The ID starts at 1
+ * and increments on each call. The returned value is not universally unique,
+ * but it is unique across the life of the PHP process and it's stable per
+ * prefix.
+ *
+ * @since 6.4
+ *
+ * @param string $prefix Optional. Prefix for the returned ID. Default empty string.
+ * @return string Incremental ID per prefix.
+ */
+function wp_unique_prefixed_id( $prefix = '' ) {
+	static $id_counters = array();
+	if ( ! isset( $id_counters[ $prefix ] ) ) {
+		$id_counters[ $prefix ] = 0;
+	}
+
+	return $prefix . (string) ++$id_counters[ $prefix ];
```

Comment:
nit: I find the incrementing inside the return to be a bit harder to read. I wonder if it might be better to increment and then return.

Reply:
It's consistent with `wp_unique_id()`. That said, I thought the same thing : separate the increment from the string build/return.

Reply:
Maybe we can refactor both in 6.5, as it is code quality. I can leave the PR ready after the RC, pointing trunk branch.

Reply:
I think what makes it less readable as compared to `wp_unique_id()` is the array lookup. To help, split the increment in 11f1cace5cb60369011f9e26e7ada5e412c70b57.